drivers/gpio/gpio_mmio32.c

```diff
@@ -41,7 +41,7 @@ static int gpio_mmio32_config(const struct device *dev,
 
 	if (flags & ~(GPIO_INPUT | GPIO_OUTPUT |
 		      GPIO_OUTPUT_INIT_LOW | GPIO_OUTPUT_INIT_HIGH |
-		      GPIO_ACTIVE_LOW)) {
+		      GPIO_ACTIVE_LOW | GPIO_OUTPUT_INIT_LOGICAL)) {
```

Comment:
Why is this driver the only one that needs this?  `GPIO_OUTPUT_INIT_LOGICAL` and all polarity-related overrides should be handled by the `gpio_pin_configure()` wrapper so this function doesn't see anything but physical configurations.

Reply:
hmm, need to double check what's going on here.

Reply:
I think the driver complains if the pin being configured is active high. In `gpio_pin_configure()` wrapper in `gpio.h` we have
```
if (((flags & GPIO_OUTPUT_INIT_LOGICAL) != 0)
    && ((flags & (GPIO_OUTPUT_INIT_LOW | GPIO_OUTPUT_INIT_HIGH)) != 0)
    && ((flags & GPIO_ACTIVE_LOW) != 0)) {
	flags ^= GPIO_OUTPUT_INIT_LOW | GPIO_OUTPUT_INIT_HIGH
		| GPIO_OUTPUT_INIT_LOGICAL;
}
```
which correctly flips `GPIO_OUTPUT_INIT_LOGICAL` flag only if pin is active low. If pin is active high the flag will be passed onto the driver.

Reply:
Yes, that's probably it.  @galak Try changing this to:
```
if (((flags & GPIO_OUTPUT_INIT_LOGICAL) != 0)
    && ((flags & (GPIO_OUTPUT_INIT_LOW | GPIO_OUTPUT_INIT_HIGH)) != 0)
    && ((flags & GPIO_ACTIVE_LOW) != 0)) {
	flags ^= GPIO_OUTPUT_INIT_LOW | GPIO_OUTPUT_INIT_HIGH;
}
flags &= ~GPIO_OUTPUT_INIT_LOGICAL;
```

Reply:
If you confirm that works with mmio32 I can do a PR, or if you update this PR I'll do the all-platform testing the change will require (along with an updated test).

Reply:
@pabigot confirm this fixes my issue.  If you have the bandwidth to send a PR that would be great.

Reply:
#32372